### PR TITLE
Clean the code a bit

### PR DESCRIPTION
**Pull request by frostming**
Do some refactor to make the code cleaner and easier to extend.

### CODE REVIEW

**Comment by yihong0618** [comment]
cool seems easily add amazon.jp. will test locally and then merge~

File: kindle.py [hunk 1]
```diff
@@ -189,13 +182,11 @@ def download_books(self):
...
             + device["deviceSerialNumber"]
         )
-        with open(os.path.join(OUT_DIR, "key.txt"), "w") as f:
+        with open(os.path.join(self.out_dir, "key.txt"), "w") as f:
             f.write(f"Key is: {device['deviceSerialNumber']}")
 
 
 if __name__ == "__main__":
-    if not os.path.exists(OUT_DIR):
-        os.mkdir(OUT_DIR)
```
Comment (yihong0618) [inline kindle.py:198]: In this way we do not need to check the exist?

**Review by yihong0618 — APPROVED** [review]
LGTM

File: kindle.py [hunk 2]
```diff
@@ -219,13 +210,18 @@ def download_books(self):
         default=100,
         help="recover-index if download failed",
     )
+    parser.add_argument(
+        "-o", "--outdir", default=DEFAULT_OUT_DIR, help="dwonload output dir"
+    )
     options = parser.parse_args()
+
+    if not os.path.exists(options.outdir):
+        os.makedirs(options.outdir)
```
Comment (frostming) [inline kindle.py:219]: I've move it here, making outdir a var that can be changed.

Comment (yihong0618) [reply kindle.py:219]: cool thanks.

**Pull request merged**